torchtitan/float8_linear.py

```diff
@@ -27,8 +44,8 @@ def build_fp8_linear(model: nn.Module, job_config: JobConfig):
     This will mutate the model inplace.
     """
     use_fp8_linear = job_config.training.fp8_linear
+    enable_fsdp_fp8_all_gather = job_config.training.enable_fsdp_fp8_all_gather
```

Comment:
discussed offline: please check if it makes sense to enable it only when `dp_degree > 1`

Reply:
added check on `parallel_dims.dp_enabled`